boards/common/iotlab/include/periph_conf_common.h

```diff
@@ -149,6 +149,8 @@ static const i2c_conf_t i2c_config[] = {
         .speed          = I2C_SPEED_NORMAL,
         .scl_pin        = GPIO_PIN(PORT_B, 6),
         .sda_pin        = GPIO_PIN(PORT_B, 7),
+        .scl_af         = GPIO_AF_OUT_OD,
```

Comment:
all those changes are useless since af cannot be changed on STM32 F1.
Normally, the scl_af and sda_af attributes are not used when the CPU is stm32f1.

Reply:
This change is needed because of CPP check, @smlng also modified the driver accordingly.

Reply:
@aabadie as @dylad wrote, I changed the driver accordingly omitting the `#ifdef` in there. However, I agree with you as this is kind of non-dynamic and fixed for stm32f1 CPUs it should not be exposed in the configuration of the board, which suggests it might be dynamic. But this is a general problem in many drivers and their config: i.e, that RIOT exposes params in configs that are actually not configurable at all. 

Nevertheless, C++ requires to init all struct attributes properly and in order, omitting one or more or changing the order will result in compile errors. Hence I had to expose the scl/sda_af in the config.